persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/DelegatingAuthorizer.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cassandra.impl;
+
+import io.stargate.auth.AuthorizationOutcome;
+import io.stargate.auth.AuthorizationProcessor;
+import io.stargate.auth.PermissionKind;
+import io.stargate.auth.entity.AccessPermission;
+import io.stargate.auth.entity.AuthorizedResource;
+import io.stargate.auth.entity.EntitySelector;
+import io.stargate.auth.entity.ImmutableAccessPermission;
+import io.stargate.auth.entity.ImmutableActor;
+import io.stargate.auth.entity.ImmutableAuthorizedResource;
+import io.stargate.auth.entity.ResourceKind;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.cassandra.auth.AuthenticatedUser;
+import org.apache.cassandra.auth.CassandraAuthorizer;
+import org.apache.cassandra.auth.DataResource;
+import org.apache.cassandra.auth.FunctionResource;
+import org.apache.cassandra.auth.IResource;
+import org.apache.cassandra.auth.Permission;
+import org.apache.cassandra.auth.RoleResource;
+import org.apache.cassandra.exceptions.RequestExecutionException;
+import org.apache.cassandra.exceptions.RequestValidationException;
+import org.apache.cassandra.stargate.exceptions.AuthenticationException;
+
+public class DelegatingAuthorizer extends CassandraAuthorizer {
+
+  private static final long PROCESSING_TIMEOUT_MILLIS =
```

Comment:
please use `Duratiion.ofMinutes(5).getMillis()` for providing the default. It will be more readable.